src/utilities.h

```diff
@@ -346,6 +346,19 @@ std::string getEntityName(const EntityPtr &entity);
  * @param entity The entity to get the owning model for.
  * @return The owning @c Model or the @c nullptr if no model owns this entity.
  */
-ModelPtr parentModel(const EntityPtr &entity);
+ModelPtr owningModel(const EntityPtr &entity);
+
+/**
+ * @brief Remove the given component for the given entity.
+ *
+ * The entity is expected to be derived from @c ComponentEntity and the
+ * component is expected to be a direct child of that entity.  The component
+ * will only be removed from the entity itself the component hierarchy of the
+ * entity will not be searched for the location of the component.
```

Comment:
... I'm not sure this makes sense? "The component will only be removed from the entity itself the component hierarchy of the entity will not be searched for the location of the component" ... and even if you add an "if" into the middle, I still don't really follow it :|

Reply:
I am resolving this conversation since this PR looks good to @nickerso and I, but @kerimoyle feel free to create an issue for this, if needed and still relevant.